dan_layer/storage_sqlite/src/sqlite_shard_store_factory.rs

```diff
@@ -712,6 +715,79 @@ impl ShardStoreTransaction<PublicKey, TariDanPayload> for SqliteShardStoreTransa
         Ok(())
     }
 
+    fn get_state_inventory(&self, start_shard: ShardId, end_shard: ShardId) -> Result<Vec<ShardId>, Self::Error> {
```

Comment:
Should we not try and combine these two functions?

The actual sqlite query is the same, we just wrap the data differently? 
Either we need to wrap it differently upstream or keep them the same and just change how we use the data?

Reply:
Refactored this piece of code.